unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/PlayerPrefsUtils/PlayerPrefsProviderLocalStorage.cs

```diff
@@ -0,0 +1,106 @@
+using System.Runtime.InteropServices;
+using UnityEngine;
+
+namespace DCL.Helpers
+{
+    /// <summary>
+    ///     Provider class to use LocalStorage instead of IndexedDB because it was not reliably saving settings.
+    ///     For all the gets, we have to check that the PlayerPref key does exist. We may lose information that was
+    ///     succesfully stored in IndexedDB. That check should only go once per user per key.
+    /// </summary>
+    public class PlayerPrefsProviderLocalStorage : IPlayerPrefsProvider
+    {
+        public int GetInt(string key) =>
+            GetInt(key, 0);
+
+        public int GetInt(string key, int defaultValue)
+        {
+            var valueGotFromLocalStorage = int.Parse(GetLocalStorageValue(key, defaultValue.ToString()));
+
+            if (valueGotFromLocalStorage.Equals(defaultValue))
+            {
+                if (PlayerPrefs.HasKey(key)) return PlayerPrefs.GetInt(key);
+            }
```

Comment:
there is so much love for nested `if`'s in our project :) I would recommend to combine them in one `if`-check by using `&&`

Reply:
Sure, sound good